### DISCUSSION TITLE
Max Boost Clock Override not recognized in Corefreq (Zen3/Vermeer)

### DISCUSSION THREAD

**Discussion by gel-crabs** [cyring/CoreFreq#377] [General]
Hi again, in the PBO2 menu in the BIOS there is Max Boost Clock Override, which allows you to overclock. However, changing the boost clock override is not recognized by CoreFreq; the max frequency does not change. Need some debug info?

(Ryzen 5 5600X)

**Comment by cyring** [comment]
I just know the Register to read CPB ratios

I can however write non boosted P-states `Max`, `1C` and `2C`

# `1C` and `2C`
1. Turbo Boost `CPB` must disabled
![2022-11-24-164136_495x222_scrot](https://user-images.githubusercontent.com/11563789/203822693-c0231bd4-41ea-4ee1-a4ea-9f4627bb249f.png)
2. Select Ratio
![2022-11-24-164223_626x322_scrot](https://user-images.githubusercontent.com/11563789/203822868-2ae5798f-3371-4875-a64d-a6201ca3e617.png)
3. Alter the whole Processor
![2022-11-24-164349_383x487_scrot](https://user-images.githubusercontent.com/11563789/203823059-c31cdede-057a-48d7-8276-8180657e918e.png)
![2022-11-24-164506_248x526_scrot](https://user-images.githubusercontent.com/11563789/203823307-97542cfd-6fda-4d0c-81fd-72feb4b17ad6.png)
4. CPB can now be reactivated
![2022-11-24-164629_302x111_scrot](https://user-images.githubusercontent.com/11563789/203823567-d1cd9519-f89f-4660-9471-a2cd9ee77fb5.png)


# `Max` (non-boosted) frequency

1. Need a non TSC clock source
`HPET` is fine
![2022-11-24-165052_637x246_scrot](https://user-images.githubusercontent.com/11563789/203824675-d047cf03-6ca3-41c6-86c6-2d1f11c59d8e.png)
2. Turbo Boost `CPB` must disabled
3. Select a new frequency ratio
![2022-11-24-165434_383x739_scrot](https://user-images.githubusercontent.com/11563789/203825248-22e14a1a-65e5-4911-a670-19cfc6738974.png)
**Check kernel log for any instability**: kernel TSC clock source may be lost.
You can switch to CoreFreq TSC which is aware about what you are doing.
![2022-11-24-165701_302x97_scrot](https://user-images.githubusercontent.com/11563789/203825704-730cfa71-542c-436b-ad9b-1a147a64215f.png)
